src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -3658,7 +3674,7 @@ private static void checkValidForByteConversion(final SmallestDataStorageKibibyt
         {
             throw new ConfigurationException(String.format("%s must be positive value < %dB, but was %dB",
```

Comment:
so, with your patch `< 0` isn't possible, so this error msg is confusing as it only happens if `valueInBytes > Integer.MAX_VALUE`, so it IS positive... its just too big

Reply:
It reiterates the rule - this is what you can give me.

Reply:
Should the error message use `<=`, given we fail for `>`?

```suggestion
            throw new ConfigurationException(String.format("%s must be positive value <= %dB, but was %dB",
```

Reply:
suggestion committed